client-app/src/common/models/search/paged-search-criteria.ts

```diff
@@ -1,16 +1,16 @@
-import { SearchQuery } from './search-query';
+import { PagedSearchQuery } from '@common/models/search/paged-search-query';
 
 /* eslint-disable-next-line @typescript-eslint/interface-name-prefix */
-export interface ISearchCriteria {
+export interface IPagedSearchCriteria {
   pageNumber?: number;
   pageSize?: number;
 }
 
-export class SearchCriteria implements ISearchCriteria {
+export class PagedSearchCriteria implements IPagedSearchCriteria {
   pageNumber?: number;
   pageSize?: number;
 
-  constructor(data?: ISearchCriteria) {
+  constructor(data?: IPagedSearchCriteria) {
     if (data) {
```

Comment:
1. If you copy data from one source to another so why couldn't you move this logic to a dedicated function with a clear name? ) It is simplify understanding and reading :)

2. quite a deep nesting of conditions, it's hard to read from the first sight. If data is initialized only if it's true, we can return from immediately if it's not. Thus we making shorter our nesting, eg:
![image](https://user-images.githubusercontent.com/12604378/78963832-eb990500-7b22-11ea-84fb-60b18bfa8ae4.png)

Reply:
1. It comes from auto-generated API models. Because we already use this approach when use them, I'll prefer to use it here too, instead of using different.
2. Agree. It's calling preconditions and it's better to use them (same as we throw exceptions in C#)